07-http_gordle/final/internal/handlers/guess/guess.go

```diff
@@ -0,0 +1,114 @@
+package guess
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+
+	"learngo-pockets/httpgordle/api"
+	"learngo-pockets/httpgordle/internal/domain"
+	"learngo-pockets/httpgordle/internal/gordle"
+	"learngo-pockets/httpgordle/internal/handlers/apiconversion"
+	"learngo-pockets/httpgordle/internal/repository"
+)
+
+type gameGuesser interface {
+	Find(domain.GameID) (domain.Game, error)
+	Update(domain.GameID, domain.Game) error
```

Comment:
This `Find` and `Update` pair are not concurrency safe.

Reply:
We will cover it as a limitation.

Reply:
If you are writing them for a web api, they must be concurrency safe... otherwise they can lead to security vulnerabilities.